File: flutter_app/lib/models/budget.dart
class Budget {
  final String id;
  final String userId;
  final String category;
  final double limit;
  final int month;
  final int year;

  Budget({
    required this.id,
    required this.userId,
    required this.category,
    required this.limit,
    required this.month,
    required this.year,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'user_id': userId,
      'category': category,
      'limit': limit,
      'month': month,
      'year': year,
    };
  }

  factory Budget.fromMap(Map<String, dynamic> map) {
    return Budget(
      id: map['id'] ?? '',
      userId: map['user_id'] ?? '',
      category: map['category'] ?? '',
      limit: (map['limit'] ?? 0).toDouble(),
      month: map['month'] ?? 0,
      year: map['year'] ?? 0,
    );
  }
}

File: backend/models/Budget.js
const mongoose = require('mongoose');

const BudgetSchema = new mongoose.Schema({
  id: { type: String, required: true, unique: true },
  user_id: { type: String, required: true, index: true },
  category: { type: String, required: true },
  limit: { type: Number, required: true },
  month: { type: Number, required: true },
  year: { type: Number, required: true }
}, { timestamps: true });

// Composite index for efficient querying by user, month, year
BudgetSchema.index({ user_id: 1, month: 1, year: 1 });

module.exports = mongoose.model('Budget', BudgetSchema);
